### PR TITLE
Fixes #18844: Add a curl/openssl wrapper for sending file to policy server without validating host

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/18844

### CODE REVIEW

**Comment by amousset** [comment]
Work in progress, only a part of the commands is implemented.

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: agent/sources/client/Makefile [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+TEST_OBJ := run.o
+TEST_EXEC := test
+
+OUR_SRCS := $(shell find $(SRC_DIR) -name "*.[ch]" ! -path "*argtable3*")
+
+# main is handled separately
+SRCS := $(shell find $(SRC_DIR) ! -path *main.c -name *.c)
+OBJS := $(patsubst $(SRC_DIR)/%,$(BUILD_DIR)/%,$(SRCS:.c=.o))
+
+DESTDIR ?= $(BUILD_DIR)/dest
```
Comment (peckpeck) [inline agent/sources/client/Makefile:19]: all make variables are overidable by command line, is think at least those 2 those should not be overiden by evironment

File: agent/sources/client/config/agent.conf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# This file configures the way the Rudder agent connects to its server
```
Comment (peckpeck) [inline agent/sources/client/config/agent.conf:1]: Add comments for default values

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: agent/sources/client/config/agent.conf [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+server = "rudder.example.com"
```
Comment (peckpeck) [inline agent/sources/client/config/agent.conf:3]: Server default value should be read from policy_server.dat

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
Still not correct

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by gpoblon** [comment]
not a reviewer but i'd have approved it, checked mem leaks, pointer references, and overall code. Was nice to read C again, I like Rust even more now.

File: agent/sources/client/config/agent.conf (hunk 3 above)
Comment (peckpeck) [reply agent/sources/client/config/agent.conf:3]: if we distribute this file as a default configuration file, we don't want to have any value here

Comment (amousset) [reply agent/sources/client/config/agent.conf:3]: indeed

**Comment by amousset** [comment]
PR updated with a new commit

**Review by peckpeck — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**